linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_kmod/rule.yml

```diff
@@ -1,7 +1,5 @@
-{{%- if product in ["rhel8", "rhel9"] %}}
+{{%- if product in ["rhel7", "rhel8", "rhel9"] %}}
     {{%- set kmod_audit="-a always,exit -F path=/usr/bin/kmod -F perm=x -F auid>=1000 -F auid!=unset -F key=privileged" %}}
-{{%- elif product in ["rhel7"] %}}
-    {{%- set kmod_audit="-w /usr/bin/kmod -p x -F auid!=unset -k module-change" %}}
```

Comment:
There are more details in this change. The rule asks to monitor the `execution` of `kmod`, like it was happening in the removed line. So, it was a rule using `File System` control.

Now it is a rule using `System Call` control. However, there is no `syscal` informed. So, my first concern is about performance. I didn't find the information to assess the impact of rules without a `syscal`. Maybe some tests are necessary to have this information. On the other hand, the man says that is efficient to inform more syscals in the same rule. Not informing any `syscal` would create the opposite effect, of making it less efficient?

The other concern, also related to the leaking of `syscals` in the rule is that the proposed rule would get much more than `kmod` execution. Besides execution, read, write and changes would also match this rule. Is that the desired effect?

I believe it will impact performance. Do you have more information about this concerns I mentioned? Have you assessed the performance impact of this change?

Reply:
> There are more details in this change. The rule asks to monitor the `execution` of `kmod`, like it was happening in the removed line. So, it was a rule using `File System` control.
> 
> Now it is a rule using `System Call` control. However, there is no `syscal` informed. So, my first concern is about performance. I didn't find the information to assess the impact of rules without a `syscal`. Maybe some tests are necessary to have this information. On the other hand, the man says that is efficient to inform more syscals in the same rule. Not informing any `syscal` would create the opposite effect, of making it less efficient?

The funcionality should be the same and with similar performance. From `man auditctl` page, `-w path` is the same as using `-F path=`:
```
       -w path
              Insert a watch for the file system object at path. You cannot insert a watch to the top level directory. This is prohibited by the kernel. Wildcards are not supported either and will generate a warning.  The  way  that
              watches work is by tracking the inode internally. If you place a watch on a file, its the same as using the -F path option on a syscall rule. If you place a watch on a directory, its the same as using the -F dir option
              on a syscall rule. The -w form of writing watches is for backwards compatibility and the syscall based form is more expressive. Unlike most syscall auditing rules, watches do not impact performance based on the  number
              of  rules  sent to the kernel. The only valid options when using a watch are the -p and -k. If you need to do anything fancy like audit a specific user accessing a file, then use the syscall auditing form with the path
              or dir fields. See the EXAMPLES section for an example of converting one form to another.

```
> 
> The other concern, also related to the leaking of `syscals` in the rule is that the proposed rule would get much more than `kmod` execution. Besides execution, read, write and changes would also match this rule. Is that the desired effect?

As the rule will still be using `-F perm=x` the scope should be the same.
From the same man page:
```
              perm        Permission  filter  for  file operations. See "-p". It can only be used on exit list. You can use this without specifying a syscall and the kernel will select the syscalls that satisfy the permissions being
                          requested.

```

> 
> I believe it will impact performance. Do you have more information about this concerns I mentioned? Have you assessed the performance impact of this change?

No, I have not assessed the performance impact of using a syscal rule without any syscal.
From the man page it seems the performance should be similar, if not identical.

Reply:
See also the example section from auditctl man page:
```
       To watch a file for changes (2 ways to express):

       auditctl -w /etc/shadow -p wa
       auditctl -a always,exit -F path=/etc/shadow -F perm=wa

       To recursively watch a directory for changes (2 ways to express):

       auditctl -w /etc/ -p wa
       auditctl -a always,exit -F dir=/etc/ -F perm=wa
```

Reply:
Thanks for clarifying my concerns. Although the man page mentions the equivalence of "File System" and "System Calls", it is still not clear if they differs in performance. I believe some tests to create evidence would be good, especially considering the number of rules we have. But I agree that the performance impact assessment is not in the scope of this PR. I would suggest you to open an issue about this. Maybe there are opportunities to improve the performance of these rules.

Regarding my second point, nevermind. I overlooked the `-F perm=x`.

Reply:
@marcusburghardt would you mind filing the issue?

Reply:
#9849